python/lsst/pipe/analysis/colorAnalysis.py

```diff
@@ -363,281 +560,1040 @@ def plotGalaxyColors(self, catalogs, filenamer, dataId):
             catalog.writeFits("gi.fits")
             shortName = "galaxy-TEST"
             self.log.info("shortName = {:s}".format(shortName))
-            self.AnalysisClass(catalog, GalaxyColor("modelfit_CModel_flux", "slot_CalibFlux_flux", "g_", "i_"),
+            self.AnalysisClass(catalog,
+                               GalaxyColor("modelfit_CModel_flux", "slot_CalibFlux_flux", "g_", "i_"),
                                "(g-i)_cmodel - (g-i)_CalibFlux", shortName, self.config.analysis,
                                flags=["modelfit_CModel_flag", "slot_CalibFlux_flag"], prefix="i_",
                                labeller=OverlapsStarGalaxyLabeller("g_", "i_"),
                                qMin=-0.5, qMax=0.5,).plotAll(dataId, filenamer, self.log)
 
-    def plotStarColors(self, catalog, filenamer, labeller, dataId, butler=None, camera=None, tractInfo=None,
-                       patchList=None, hscRun=None):
+    def plotStarColors(self, principleColCats, byFilterCats, filenamer, labeller, dataId, butler=None,
+                       camera=None, tractInfo=None, patchList=None, hscRun=None):
+        mags = {ff: -2.5*np.log10(byFilterCats[ff]["base_PsfFlux_flux"]) for ff in byFilterCats}
+        color = lambda c1, c2: (mags[c1] - mags[c2])
         unitStr = "mmag" if self.config.toMilli else "mag"
         for col, transform in self.config.transforms.items():
```

Comment:
Most of this method is a loop over col's. A "plotStarColor" method would be welcome if it makes sense